### PR TITLE
expose ArrayField type

### PR DESCRIPTION
**Pull request by bluebill1049**
expose: `ArrayField`  type

https://github.com/react-hook-form/react-hook-form/discussions/2174#discussioncomment-38042

- [ ]  update doc

### CODE REVIEW

File: src/types/index.ts [hunk 1]
```diff
@@ -9,5 +9,6 @@ export {
   ValidationRules,
   FieldError,
   Field,
+  ArrayField,
```
Comment (kotarella1110) [inline src/types/index.ts:12]: I think better rename to `ArrayFieldState`.
What do you think ?

Comment (bluebill1049) [reply src/types/index.ts:12]: how about `ArrayFields`?

Comment (kotarella1110) [reply src/types/index.ts:12]: `ArrayField` type is not array type.
`ArrayField` or `ArrayFieldState` is better than `ArrayFields` 👍

Comment (bluebill1049) [reply src/types/index.ts:12]: you are right. shall we just keep `ArrayField` then? it's a represent the shape of FieldArray item.

Comment (bluebill1049) [reply src/types/index.ts:12]:
```
const toniosCat: ArrayField<{ test: string}> = {
  test: 'test',
  id: 'xxx'
};
```

looks good as above :)

Comment (kotarella1110) [reply src/types/index.ts:12]: let's keeep `ArrayField` 👍

**Review by kotarella1110 — APPROVED** [review]

File: src/types/index.ts (hunk 1 above)
Comment (bluebill1049) [reply src/types/index.ts:12]: thanks for the review :)

**Pull request merged**